src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,99 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  addComment: (comment: CommentData) => void;
+  commentAdditionError: boolean;
+  commentAdditionLoading: boolean;
+};
+
+export const NewCommentForm: React.FC<Props> = React.memo(({
+  postId,
+  addComment,
+  commentAdditionError,
+  commentAdditionLoading,
+}) => {
+  const [userName, setUserName] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [userComment, setUserComment] = useState('');
+  const [userRequired, setUserRequired] = useState(false);
+  const [userEmailRequired, setUserEmailRequired] = useState(false);
+  const [userCommentRequired, setUserCommentRequired] = useState(false);
```

Comment:
`isUserRequired` etc
Please fix all boolean variable names
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig